decidim-assemblies/app/permissions/decidim/assemblies/permissions.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Assemblies
+    class Permissions
+      def self.allows?(user, action, context = {})
+        new(user, action, context).allowed?
+      end
+
+      def initialize(user, action, context = {})
+        @user = user
+        @action = action
+        @context = context
+      end
+
+      def allowed?
+        return true if action[:scope] == "public"
```

Comment:
There's the risk that we don't access `:scope` properly, and that `public` isn't a string but a symbol. I can see how to tackle this in two ways:

* Constructing an `action` object when checking permissions. That object could normalize its fields and structure, providing some level of sanity.
* Implementing methods in this class (probably its superclass if we create one) to access them. Don't like this, too brittle.

Reply:
When the hash is constructed in the controller, the values are casted to a String there.

I don't understand this part, though:

> There's the risk that we don't access `:scope` properly

Reply:
I mean that `scope` can be accessed as a string or symbol mistakenly, while as a method call doesn't have that drawback.

Reply:
Done, I've added the `Decidim::PermissionAction` that encapsulates this logic. Also, I've moved all values to symbols by convenience